src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java

```diff
@@ -66,6 +67,16 @@
     @SqlUpdate("DELETE FROM datasetproperty WHERE datasetid = :datasetId")
     void deleteDatasetPropertiesByDatasetId(@Bind("datasetId") Integer datasetId);
 
+    @SqlUpdate("INSERT INTO dataset_audit (datasetid, changeaction, modifiedbyuser, modificationdate, objectid, name, active) VALUES (:dataSetId, :action, :user, :date, :objectId, :name, :active )")
+    @GetGeneratedKeys
+    Integer insertDataSetAudit(@BindBean DataSetAudit dataSets);
+
+    @SqlUpdate("DELETE FROM dataset_user_association WHERE datasetid = :datasetId")
+    void deleteUserAssociationsByDatasetId(@Bind("datasetId") Integer datasetId);
+
+    @SqlUpdate("DELETE FROM consentassociations WHERE datasetid = :datasetId")
+    void deleteConsentAssociationsByDataSetId(@Bind("datasetId") Integer datasetId);
```

Comment:
These queries were originally in other DAOs. These were moved to here so they can all be wrapped in a single transaction block.